articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md

```diff
@@ -0,0 +1,159 @@
+> > [!div renderon="portal" id="makechanges" class="nextstepaction"]
+> > [Make these changes for me]()
+>
+> > [!div id="appconfigured" class="hidden"]
+> > ![Already configured](media/active-directory-windesktop/checkmark.png) Your application is configured with these attributes
+
+> [!NOTE]
+> If you use Visual Studio, the Redirect URL will be set to to *http:<span/>//localhost:30662/* as it is configured in the code sample's project. If you use Python or any other web server, set redirect URL to *http://<span/>localhost:30662/*, run the following command line:
+>> ```bash
+>> python -m http.server 30662
+>> ```
+
+### Step 2: Download your web server or project
+
+- [Download the Visual Studio project](https://github.com/Azure-Samples/active-directory-javascript-graphapi-v2/archive/VisualStudio.zip)
+- [Download the core project files - for a local web server, such as Python](https://github.com/Azure-Samples/active-directory-javascript-graphapi-v2/archive/core.zip)
+
+### Step 3: Configure your JavaScript SPA
+
+> [!div renderon="docs"]
+> Edit `msalconfig.js` and replace `Enter_the_Application_Id_here` with the Application ID your application:
+
+> [!div class="sxs-lookup" renderon="portal"]
+> Edit `msalconfig.js` and replace msalconfig with:
+
+```javascript
+var msalconfig = {
+    clientID: "Enter_the_Application_Id_here",
+    redirectUri: location.origin
+};
+```
+
+## More Information
+
+### *msal.js*
+
+MSAL is the library used to sign in users and request tokens used to access an API protected by Microsoft Azure Active Directory. The Quickstart's *index.html* contains reference to the library:
+
+```html
+<script src="https://secure.aadcdn.microsoftonline-p.com/lib/0.1.1/js/msal.min.js"></script>
+````
+
+### MSAL initialization
+
+The quickstart code also shows how you how you initialize the library:
+
+```javascript
+var userAgentApplication = new Msal.UserAgentApplication(msalconfig.clientID, null, loginCallback, {
+    redirectUri: msalconfig.redirectUri
+});
+```
+
+> |Where  |  |
+> |---------|---------|
+> |ClientId     |Application Id from the application registered in the Azure Portal|
+> |loginCallBack | Name of a callback method called after the authentication|
+> |redirectUri     |URL where users are sent after authentication against Azure AD v2 Endpoint|
+
+### Sign-in users
+
+Below how you sign in users:
+
+```javascript
+userAgentApplication.loginRedirect(graphAPIScopes);
+```
+
+> |Where  |  |
+> |---------|---------|
+> |graphAPIScopes     | (Optional) Contains scopes being requested at login time (i.e. `{ "user.read" }` for Microsoft Graph or `{ "api://<Application ID>/access_as_user" }` for custom Web APIs)|
+
+> [!TIP]
+> Alternativelly you may want to use `loginPopup` method to display a popup window for sign-in the user.
+
+### Requesting tokens
+
+Msal has two methods to used acquire tokens `acquireTokenRedirect` (or `acquireTokenPopup`) and `acquireTokenSilent`:
```

Comment:
Can we reword it? Even though we say 2 methods there are actually 3 and this can be confusing. We may categorize it as  'Acquire Token Interactive APIs' (acquireTokenRedirect and acquireTokenPopup ) and 'Acquire Token Silent API' (acquireTokenSilent)